nevergrad/benchmark/experiments.py

```diff
@@ -570,6 +621,13 @@ def yabbob(seed: tp.Optional[int] = None, parallel: bool = False, big: bool = Fa
         for num_blocks in [1]
         for d in ([100, 1000, 3000] if hd else [2, 10, 50])
     ]
+    assert constraints < 8, "We have only four possible constraints."
+    for func in functions:
+        constraints_list = [_positive_sum_float(), _positive_diff_float(), _positive_second_diff_float(), _ball_float(),
+            _positive_sum(), _positive_diff(), _positive_second_diff(), _ball()]
+        for u in range(8):
+            if constraints > u and constraints <= u+4:
+                func.parametrization.register_cheap_constraint(constraints_list[u])
```

Comment:
It's much better, but I still don't get why there is this weird u, u+4 case, can you explain so that we find a clearer option?

Reply:
We just want to have various combinations of boolean and float constraints. With this simple if we get 8 different combinations and all cases: all boolean, all float, and mixed.

Reply:
I'm still unclear on what that does, especially since the order in the nested for loops creating the constraints has never been intuitive to me (is a  sequence of bool = True then False and so on? or all True, then all False?)
Wouldn't that be more explicit to use the name and the boolean directly, and maybe two for loops, instead of the objects themselves ? not sure, since I still don't get it
Also, is it expected that u == 0 has no constraint?

Reply:
I have checked the order of the nested "for for" in the list for being sure that it does what I want.
I have, on purpose, modified the code for not multiplying the number of functions - for the moment we can still afford modifications (I'd like the problems to be somehow stable over 2021).

Please note that for lists, list comprehension are deterministic and the order matters, as explained in https://stackoverflow.com/questions/1198777/double-iteration-in-list-comprehension, "nested" for are in some sense horizontal.

Reply:
My last modification was not very good for diversity of cases, I've modified back while keeping the new style of that code.